src/Autocomplete/src/Form/AutocompleteChoiceTypeExtension.php

```diff
@@ -91,6 +100,41 @@ public function finishView(FormView $view, FormInterface $form, array $options):
         $view->vars['attr'] = $attr;
     }
 
+    private function getUrlWithExtraOptions(string $url, array $extraOptions): string
+    {
+        $this->validateExtraOptions($extraOptions);
+
+        $extraOptions[self::CHECKSUM_KEY] = $this->checksumCalculator->calculateForArray($extraOptions);
+        $extraOptions = base64_encode(json_encode($extraOptions));
+
+        return sprintf(
+            '%s%s%s',
+            $url,
+            $this->hasUrlParameters($url) ? '&' : '?',
+            http_build_query(['extra_options' => $extraOptions]),
+        );
+    }
+
+    private function hasUrlParameters(string $url): bool
+    {
+        $parsedUrl = parse_url($url);
+
+        return isset($parsedUrl['query']);
+    }
+
+    private function validateExtraOptions(array $extraOptions): void
+    {
+        foreach ($extraOptions as $option) {
+            if (!\is_scalar($option) && !\is_array($option) && null !== $option) {
+                throw new \InvalidArgumentException(sprintf('Extra option "%s" must be a scalar value, an array or null. Got "%s".', $option, get_debug_type($option)));
```

Comment:
Would that not create an error if $option is a resource or an non stringable object ? 

I'd use keys there i think

Reply:
The requirement was to allow setting only scalar values for the extra options, and these lines validate this requirement 🤔.

Reply:
What i meant is the sprintf will throw an error if the first %s is non stringable

Reply:
> Scalar variables are those containing an int, float, string or bool. Types array, object, resource and null are not scalar.

This `if` statement won't pass anything else then `int`, `float`, `string`, `bool` nor `array`.

Reply:
https://3v4l.org/Wf0ke

Reply:
This line will throw an error if $option is NOT scalar and NOT array and NOT null ;) 

```php
sprintf('Extra option "%s" must[...]. Got "%s".', $option, get_debug_type($option));
```

Reply:
Damn :D I was blind :D. Thanks!

Reply:
Don't be :) I should have tested earlier... and suggested a fix...  that would have been more constructive :)